pandas/_libs/lib.pyx

```diff
@@ -584,8 +585,12 @@ def array_equivalent_object(left: object[:], right: object[:]) -> bool:
                     return False
             elif (x is C_NA) ^ (y is C_NA):
                 return False
-            elif not (PyObject_RichCompareBool(x, y, Py_EQ) or
-                      (x is None or is_nan(x)) and (y is None or is_nan(y))):
+            elif not (
+                PyObject_RichCompareBool(x, y, Py_EQ)
+                or is_matching_na(x, y)
```

Comment:
instead of this an you add a keyword to is_matching_na? (actually would make it easier to deprecate then), maybe `allow_equivalent=True`?

Reply:
sure

Reply:
updated + greenish